src/transaction.spec.ts

```diff
@@ -40,6 +40,28 @@ describe("test transaction construction", async () => {
         assert.deepEqual(transaction.getSignature(), Buffer.from([]));
     });
 
+    it("create transaction from transaction next", async () => {
+        const nextTransaction = new TransactionNext({
```

Comment:
I'd extract the transaction payload and use it as an argument to the TransactionNext object and also in each assert.
It would help a lot when you write the asserts to avoid typos, and mismatched values or in case you will extend this test in the future.
This is just a recommendation, nothing wrong with your current implementation.

```
const transactionPayload = {
    sender: "erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th",
    receiver: "erd1spyavw0956vq68xj8y4tenjpq2wd5a9p2c6j8gsz7ztyrnpxrruqzu66jx",
    gasLimit: 56000,
    value: "1000000000000000000",
    data: Buffer.from("test"),
    chainID: "T"
}

const nextTransaction = new TransactionNext(transactionPayload);

const transaction = Transaction.fromTransactionNext(nextTransaction);
assert.deepEqual(transaction.getSender(), Address.fromBech32(transactionPayload.sender));
assert.deepEqual(transaction.getReceiver(), Address.fromBech32(transactionPayload.receiver));
assert.equal(transaction.getGasLimit().valueOf(), transactionPayload.gasLimit);
...
```

Reply:
Done! Extracted the payload and used it as above. Thanks for the suggestion.